re_data_ui/src/pages/GraphView.tsx

```diff
@@ -51,33 +64,40 @@ const generateGraph = (overview: OverviewData) => {
   const dbtNodes = overview.graph.nodes;
   const dbtSources = overview.graph.sources;
 
-  Object.entries(dbtNodes).forEach(([model, details]) => {
-    const [resource, packageName, modelName] = model.split('.');
-    if (resource !== 'test' && packageName !== 're_data') {
+  const allNodes = { ...dbtNodes, ...dbtSources };
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  Object.entries(allNodes).forEach(([_, details]) => {
+    if (supportedResourcesTypes.includes(details.resource_type) && details.package_name !== 're_data') {
       const modelId = `${details.database}.${details.schema}.${details.name}`.toLowerCase();
       const node: VisNode = {
         id: modelId,
-        label: modelName,
+        label: details.name,
         shape: 'box',
+        color: {
+          background: resourceTypeColors[details.resource_type],
+        }
```

Comment:
At this array length, doing `Array.includes()` might not be an issue but I think we should go with using a set so we have constant lookups.

Reply:
done, thanks